extensions/amp-ad-network-doubleclick-impl/0.1/safeframe-host.js

```diff
@@ -340,11 +340,11 @@ export class SafeframeHostApi {
       // AMP's built in resize methodology that we use only allows expansion
       // to the right and bottom, so we enforce that here.
       'allowedExpansion_r': viewportSize.width -
-          iframeBox.width,
+          iframeBox.right,
       'allowedExpansion_b': viewportSize.height -
-          iframeBox.height,
-      'allowedExpansion_t': 0,
-      'allowedExpansion_l': 0,
+          iframeBox.bottom,
+      'allowedExpansion_t': iframeBox.top,
```

Comment:
I don't think it's possible to expand to the top or to the left because resizing is done based on changing height/width attributes (the position of the element is unchanged).

I'm not as familiar with AMP runtime. @bradfrizzell  to confirm.

Reply:
That's correct, expansion in AMP effectively only supports expanding right / down. Depending on the css of the element, this can effectively be expanding left, down, or right (i.e. if the element is centered, than expanding right by 50 will have the effect of making the element expand by 25 to the right and left)

Reply:
So I understand that may be how expansion is implemented currently, but unfortunately this implementation removes the ability to actually determine the creatives location on the page through geom, which I think we can agree is the fundamental use of geom. 

Here are samples that explain the problem. 
In the current implementation geom is identical no matter where the creative is on the screen:

![brad-off-page](https://user-images.githubusercontent.com/9314256/37158867-6f5ca752-22ba-11e8-95b8-539e675e44da.png)
![brad-bottom](https://user-images.githubusercontent.com/9314256/37158868-6f6648e8-22ba-11e8-8145-89ce1a991b1d.png)
![brad-top](https://user-images.githubusercontent.com/9314256/37158869-6f7061e8-22ba-11e8-9d3f-98bdf5d3b97d.png)

For our creatives to work we need to be able to determine the location we are on the screen. 
We are able to do this in dfp-safeframes:
![safeframes-top](https://user-images.githubusercontent.com/9314256/37158913-87992476-22ba-11e8-8eb2-67d981fd93f2.png)
![safeframes-bottom](https://user-images.githubusercontent.com/9314256/37158914-87a3903c-22ba-11e8-81a9-a1b82a178c48.png)
![safeframes-off-page](https://user-images.githubusercontent.com/9314256/37158915-87b29dca-22ba-11e8-8870-c8112c14a8ca.png)

And my attempt was to mimic this behavior to allow us to determine our location on the screen:
![mine-top](https://user-images.githubusercontent.com/9314256/37158948-9a93b514-22ba-11e8-90f0-c07a1a5ce5f0.png)
![mine-bottom](https://user-images.githubusercontent.com/9314256/37158949-9a9d872e-22ba-11e8-8eff-8ee9ba03a09d.png)
![mine-off-page](https://user-images.githubusercontent.com/9314256/37158951-9cccc12c-22ba-11e8-8e09-b4953e4bb106.png)

Based on the current implementation it is impossible for our creatives to function because the given data will not let us determine anything beyond the fact that we are in view with yiv. Our creatives require something equivalent to a boundingClientRect, which we are able to calculate in the dfp-safeframes implementation and my implementation. 

I did notice that my implementation differs in returning negative numbers while dfp safeframes returns 0 when partially offscreen. I will fix this if we decide to move forward with changing this implementation. 

Let me know how you want to move forward with this.